ozaria/site/components/interactive/PageInteractive/index.vue

```diff
@@ -31,32 +31,49 @@ module.exports = Vue.extend({
     'insert-code': insertCodeComponent,
     'draggable-statement-completion': draggableStatementCompletionComponent
   },
+  watch : {
+    interactiveIdOrSlug: async function() {
+      try {
+        await this.getInteractiveData()
+      } catch (err) {
+        console.error("Error:", err)
+        return noty({ text: 'Error occured in creating interactives data.', type: 'error', timeout: '2000' })
+      }
+    }
+  },
   async created() {
     if (!me.hasInteractiveAccess()) {
       alert('You must be logged in as an admin to use this page.')
       return application.router.navigate('/', { trigger: true })
     }
     try {
-      this.interactive = await getInteractive(this.interactiveIdOrSlug)
-      this.interactiveType = this.interactive.interactiveType
-      if (!this.interactiveType) {
-        console.error("Interactive type is not set for the interactive", this.interactiveIdOrSlug)
-        noty({ text: 'Interactive type is not set for the interactive', type: 'error', timeout: '2000' })
-        return
-      }
-      const getSessionOptions = {
-        introLevelId: this.introLevelId,
-        courseInstanceId: this.courseInstanceId
-      }
-      this.interactiveSession = await getSession(this.interactiveIdOrSlug, getSessionOptions )
+      await this.getInteractiveData()
     } catch (err) {
       console.error("Error:", err)
-      noty({ text: 'Error occured in getting interactives data.', type: 'error', timeout: '2000' })
+      return noty({ text: 'Error occured in creating interactives data.', type: 'error', timeout: '2000' })
     }
   },
   methods: {
     onCompleted() {
       this.$emit('completed')
+    },
+    async getInteractiveData() {
+      try {
+        this.interactive = await getInteractive(this.interactiveIdOrSlug)
+        this.interactiveType = this.interactive.interactiveType
+        if (!this.interactiveType) {
+          return Promise.reject("Interactive type is not set for the interactive " + this.interactiveIdOrSlug)
+        }
+        if (!me.isSessionless()) {
```

Comment:
Can we add a quick comment about why we need this check here?

Reply:
This is bascially to check if we need to handle the session, or not. For example: teachers playing through a level do not have a session (their progress is not saved), hence this check. Will add a comment for the same.